packages/rum-core/src/common/observers/page-clicks.js

```diff
@@ -96,3 +83,34 @@ function getTransactionMetadata(target) {
 
   return metadata
 }
+
+// builds the name of a transaction from the given target
+// Strategy: use custom html attribute 'data-transaction-name' - otherwise fall back to "tagname" + "name"-Attribute
+function buildTransactionName(target) {
+  // Verify if a custom transaction name has been defined
+  const dtName = findCustomTransactionName(target)
+  if (dtName) {
+    return dtName
+  }
+
+  const tagName = target.tagName.toLowerCase()
+  // "tagname" + "name"-Attribute
+  const name = target.getAttribute('name')
+  if (!!name) {
+    return `${tagName}["${name}"]`
+  }
+
+  // Just use the tagName
+  return tagName.toLowerCase()
+}
+
+function findCustomTransactionName(target) {
+  if (target.closest) {
+    // Leverage closest API to traverse the element and its parents
+    const element = target.closest('[data-transaction-name]')
```

Comment:
How far do we want to traverse? We should at-least restrict to a known tags that can cause the click effect to be on safer side. 
* Button
* Anchor tags

Doing this for every observe on click is expensive and not worth it.

Reply:
Vignesh and I had a chat offline:

We will perform the following optimization:

Instead of 
```target.closest('[data-transaction-name]')```

We will do:

```target.closest('a[data-transaction-name], button[data-transaction-name]')```

This way we make sure the API just looks for interactive selectors and ignore the rest of them.